versions.tf

```
@@ -19,11 +19,11 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = "~> 3.44"
+      version = ">= 3.44, < 5.0"
     }
     google-beta = {
       source  = "hashicorp/google-beta"
-      version = "~> 3.44"
+      version = ">= 3.44, < 5.0"
```

Can we still allow 3.42 or do we need 3.44+?
```suggestion
      version = ">= 3.42, < 5.0"
```

---

I think we need to keep >= 3.44, as starting with [3.44 google_sql_database_instance](https://registry.terraform.io/providers/hashicorp/google/3.44.0/docs/resources/sql_database_instance) requires deletion_protection=false to destroy the instance.  However it appears the deletion_protection= argument isn't permitted by earlier versions.

---

Yup that sounds good, we can constrain to 3.44